### PR TITLE
Smb implementation

### PR DESCRIPTION
**Pull request by duartemateus**

### CODE REVIEW

**Comment by duartemateus** [comment]
Wonder if I should default the port on smb to 445 as I'm yet to have any luck with port 139.

File: util/ident.rb [hunk 1]
```diff
@@ -13,41 +11,38 @@ def print_debug(string)
...
     ###
     opts = Slop.parse do |o|
-
       # url input
-      o.string "-u", "--uri", "a uri to check (supported portocols: " +
-                              "(dns, elasticsearch, ftp, http, https, imap, mongodb, mysql, pop3, redis, smtp, snmp, telnet, amqp) " +
-                              "ex: http://intrigue.io"
-      o.string "-f", "--file", "a file of urls, one per line"
+      o.string '-u', '--uri', 'a uri to check (supported portocols: ' +
+                              '(dns, elasticsearch, ftp, http, https, imap, mongodb, mysql, pop3, redis, smtp, snmp, telnet, amqp, smb) ' +
```
Comment (jcran) [inline util/ident.rb]: @duartemateus minor but best if this is alphabetized.

File: lib/smb/smb.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          out << "Protocol: #{client.negotiate}"
+          client_auth = client.authenticate
+          out << "Description: #{client_auth.description}"
+          out << "Connection Status: #{client_auth.name}"
+        rescue StandardError => e
+          out = "Failed to connect to #{ip}: #{e.message}"
+        end
+
+        begin
+          out << "Native OS: #{client.peer_native_os}" if client.peer_native_os
```
Comment (jcran) [inline lib/smb/smb.rb]: @duartemateus Any reason not to use a hash for this?

Comment (duartemateus) [reply lib/smb/smb.rb]: Not any good reason no haha. I'll make the change and also create appropriate grabber functions for each key. I'll give you a nudge when it's complete, shouldn't take long

Comment (duartemateus) [reply lib/smb/smb.rb]: https://github.com/intrigueio/intrigue-ident/pull/105/commits/7c4517e7bc771f6812b3c49c4a97d8e8795f450c 

@jcran

Comment (jcran) [reply lib/smb/smb.rb]: 🔥

**Pull request merged**